Source/RapyutaSimulationPlugins/Private/Sensors/RRROS2EntityStateSensorComponent.cpp

```diff
@@ -0,0 +1,59 @@
+// Copyright 2020-2021 Rapyuta Robotics Co., Ltd.
+
+#include "Sensors/RRROS2EntityStateSensorComponent.h"
+
+URRROS2EntityStateSensorComponent::URRROS2EntityStateSensorComponent()
+{
+
+    SensorPublisherClass = URRROS2EntityStatePublisher::StaticClass();
+}
+
+void URRROS2EntityStateSensorComponent::BeginPlay()
+{
+    SetReferenceActor(ReferenceActorName);
+    Super::BeginPlay();
+}
+
+void URRROS2EntityStateSensorComponent::SetReferenceActor(const FString& InName)
+{
+    ReferenceActor = URRGeneralUtils::GetActorByName(GetWorld(), InName);
+    ReferenceActorName = InName;
+}

```

Comment:
Just a note, since `GetActorByName` is quite slow, if users could configure `ReferenceActor` directly (not sure possible in the Editor Component's details panel?), guess it would be more straightforward/robust than letting them fill in a `ReferenceActorName`.

Reply:
yes.  if you use UE and BP, it should be set directory. I'll create simillar function which get AActor* as arg.
Intension is to use this function for ROS(they can not provide Actor pointer directly)